### DISCUSSION TITLE
system weirdness when running afl++

### DISCUSSION THREAD

**Discussion by asarubbo** [AFLplusplus/AFLplusplus#1613] [General]
Starting as discussion because I didn't find the cause of the issue.

When I run afl++ (in a chroot, ftr) [emerge](https://wiki.gentoo.org/wiki/Emerge) start to become crazy.

No need to copy-paste again the story, because I have already opened an issue against portage: https://bugs.gentoo.org/890305

Do you know a bit more about this topic? I don't know whether the issue is but is obviously something unexpected.

Thanks

**Comment by andreafioraldi** [comment]
check if something in afl-system-config is in conflicy with emerge

**Reply by asarubbo** [reply]
```
~ $ afl-system-config 
This reconfigures the system to have a better fuzzing performance.
WARNING: this reduces the security of the system!

Settings applied.

It is recommended to boot the kernel with lots of security off - if you are running a machine that is in a secured network - so set this:
  /etc/default/grub:GRUB_CMDLINE_LINUX_DEFAULT="ibpb=off ibrs=off kpti=0 l1tf=off mds=off mitigations=off no_stf_barrier noibpb noibrs pcid nopti nospec_store_bypass_disable nospectre_v1 nospectre_v2 pcid=on pti=off spec_store_bypass_disable=off spectre_v2=off stf_barrier=off srbds=off noexec=off noexec32=off tsx=on tsx_async_abort=off arm64.nopauth audit=0 hardened_usercopy=off ssbd=force-off"

If you run fuzzing instances in docker, run them with "--security-opt seccomp=unconfined" for more speed.
```

**Comment by vanhauser-thc** [comment]
maybe the chroot/namespace has limits on the shared memory that afl++ aquires?

**Reply by asarubbo** [reply]
Chroot is mounted like:

```
 mount -o bind /dev "${CHROOT_DIR}"/dev
 mount -o bind /dev/pts "${CHROOT_DIR}"/dev/pts
 mount -o bind /dev/shm "${CHROOT_DIR}"/dev/shm
 mount -o bind /proc "${CHROOT_DIR}"/proc
 mount -o bind /sys "${CHROOT_DIR}"/sys

chroot "${CHROOT_DIR}" /bin/bash
```

Do you see possible problems with that?

**Reply by vanhauser-thc** [reply]
yes that is what causes the problems. maybe removing just the /dev/shm is already enough to make this work

**Reply by asarubbo** [reply]
I remounted everything by skip shm but I see no differences

**Reply by vanhauser-thc** [reply]
then try mounting nothing, or just mount them directly, not as bind (mount -t procs /proc within the chroot or how it is done)

**Reply by asarubbo** [reply]
already tried all combo without success :/

**Reply by vanhauser-thc** [reply]
sorry I know too little about gentoo and what emerge etc. do to recommend anything. I either fuzz native (temporary outputs on a ramdisk if necessary) or in a docker container.
